crates/humanode-runtime/src/lib.rs

```diff
@@ -1429,6 +1424,21 @@ impl_runtime_apis! {
         }
 
         fn gas_limit_multiplier_support() {}
+
+        fn pending_block(
+            xts: Vec<<Block as BlockT>::Extrinsic>,
+        ) -> (Option<pallet_ethereum::Block>, Option<Vec<TransactionStatus>>) {
+            for ext in xts {
+                let _ = Executive::apply_extrinsic(ext);
+            }
+
+            Ethereum::on_finalize(System::block_number() + 1);
+
+            (
+                pallet_ethereum::CurrentBlock::<Runtime>::get(),
+                pallet_ethereum::CurrentTransactionStatuses::<Runtime>::get()
+            )
```

Comment:
Shouldn't this go through a central block runtime interface? Otherwise, this seems like we are missing initializers/finalizers...

Reply:
The changes are based on 
- https://github.com/moonbeam-foundation/moonbeam/blob/ef76ba7b98dab867c17579be68e8bca5bdf6e688/runtime/common/src/apis.rs#L443
- https://github.com/polkadot-evm/frontier/pull/1048

Looks working based on tests - https://github.com/moonbeam-foundation/frontier/blob/tgm-pending-block/ts-tests/tests/test-block.ts

I guess we should add more e2e tests in the same way for our rpc eth part.

Reply:
Added the test to verify that pending block is returned - https://github.com/humanode-network/humanode/pull/848/commits/5268b4feb9386fae19164755714956fcb5d3ee36.